test-network-function/certification/suite.go

```diff
@@ -87,25 +87,32 @@ func waitForCertificationRequestToSuccess(certificationRequestFunc func() bool,
 	return isCertified
 }
 
-//nolint:dupl
 func testContainerCertificationStatus() {
 	// Query API for certification status of listed containers
 	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestContainerIsCertifiedIdentifier)
 	ginkgo.It(testID, ginkgo.Label(testID), func() {
 		env := configpkg.GetTestEnvironment()
-		containersToQuery := env.Config.CertifiedContainerInfo
-
-		if len(containersToQuery) == 0 {
+		containersToQuery := make(map[configsections.CertifiedContainerRequestInfo]bool)
+		for _, c := range env.Config.CertifiedContainerInfo {
+			containersToQuery[c] = true
+		}
+		if env.Config.CheckDiscoveredContainerCertificationStatus {
+			const redhatContainerCatalogRegistry = "https://catalog.redhat.com/software/containers/"
```

Comment:
What is the purpose of redhatContainerCatalogRegistry? the url does not look like a image registry.

Reply:
didn't you say everything under that was certified? :smile:

Reply:
This is the catalog for all certified containers, correct (https://catalog.redhat.com/software/containers/explore). But I believe that here we need the registry, similar to : https://catalog.redhat.com/software/containers/cp/ibm-eventstreams-schema-registry/5f583d17ac3db90370a1c55a?container-tabs=gti

Reply:
Not sure what you mean. We used this one for a while and the url has the prefix:

https://catalog.redhat.com/software/containers/rhel8/support-tools/5ba3eaf9bed8bd6ee819b78b

Reply:
hmm, now i see that this url doesn't use @ for digest.

Reply:
ok, i see that the real url should be like this:

registry.access.redhat.com/rhel8/support-tools@sha256:e1ae34233f8aadbe5f5b026eadfbb6c24d30b47d2e8cc612731d074bc2ad7d00

Reply:
yes, correct

Reply:
My mistake. It was actually "registry.redhat.io/rhel8/support-tools:latest".

Reply:
Maybe i will just get rid of this check? You guys can optimize it later once you confirm the certification status of the images in that registry in general.

Reply:
yes I think we can remove it. I did a quick check on the container entries in the catalog, we can at least check based on platform type (x86, arm, ..) and docker version to improve the error message

Reply:
Done.